lib/vbox/dialog.go

```diff
@@ -8,67 +8,32 @@ import (
 
 	"github.com/pborman/uuid"
 	"github.com/xshellinc/tools/constants"
+	"github.com/xshellinc/tools/dialogs"
 )
 
 // Virtualbox dialogs
 func onoff() OnOff {
-
-	var inp int
 	var a = []string{"on", "off"}
 
-	for {
-		for i, e := range a {
-			fmt.Printf("\t[\x1b[34m%d\x1b[0m] - \x1b[34m%s\x1b[0m \n", i, e)
-		}
-		fmt.Print("[+] Please select a number: ")
-
-		_, err := fmt.Scanf("%d", &inp)
-		if err != nil || inp < 0 || inp >= len(a) {
-			fmt.Println("[-] Invalid user input")
-			continue
-		}
-
-		return OnOff(inp == 0)
-	}
+	n := dialogs.SelectOneDialog("Please select a number:", a)
+	return OnOff(n == 0)
 }
 
 // NameDialog asks for VM name
 func (v *Config) NameDialog() {
-
-	var inp string
-
 	if v.Name != "" {
 		fmt.Printf("[+] Your VB name set to \x1b[34m%s\x1b[0m: \n", v.Name)
-		fmt.Print("[+] Would you like to change virtual machine name?(\x1b[33my/yes\x1b[0m OR \x1b[33mn/no\x1b[0m):")
 	} else {
 		v.Name = uuid.New()
 		fmt.Printf("[+] Your VB name is generated \x1b[34m%s\x1b[0m: \n", v.Name)
-		fmt.Print("[+] Would you like to change virtual machine name?(\x1b[33my/yes\x1b[0m OR \x1b[33mn/no\x1b[0m):")
 	}
-Loop:
-	for {
-		fmt.Scanln(&inp)
-		if strings.EqualFold(inp, "y") || strings.EqualFold(inp, "yes") {
-			for {
-				fmt.Print("[+] Enter name:")
-				inp = ""
-				fmt.Scanln(&inp)
-				if strings.EqualFold("", inp) {
-					fmt.Println("[-] Invalid user input")
-					continue
-				}
-
-				v.Name = inp
-				break Loop
-			}
-		} else if strings.EqualFold(inp, "n") || strings.EqualFold(inp, "no") {
-			break
-		} else {
-			fmt.Print("[-] Unknown user input. Please enter (\x1b[33my/yes\x1b[0m OR \x1b[33mn/no\x1b[0m)")
-		}
+
+	if dialogs.YesNoDialog("Would you like to change virtual machine name?") {
+		v.Name = dialogs.GetSingleAnswer("Enter name: ", []dialogs.ValidatorFn{dialogs.EmptyStringValidator})
 	}
 }
 
+// @todo check if needed
```

Comment:
__[golint]__
* [golint] comment on exported method Config.DescriptionDialog should be of the form "DescriptionDialog ..."

<!-- Comment by [SideCI](https://sideci.com) -->